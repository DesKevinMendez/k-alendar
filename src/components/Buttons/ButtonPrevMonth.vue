<template>
  <button type="button" class="k-alendar-navegation-prev">&larr;</button>
</template>

<script setup lang="ts"></script>

<style scoped lang="postcss">
  .k-alendar-navegation-prev {
    @apply px-2 py-1 border border-gray-200 rounded-l-md rounded-r-none w-16
    lg:p-2
    dark:border-slate-600;
  }
</style>
