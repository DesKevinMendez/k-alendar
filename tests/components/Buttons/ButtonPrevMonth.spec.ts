import ButtonPrevMonth from '@/components/Buttons/ButtonPrevMonth.vue';
import { mount, VueWrapper } from '@vue/test-utils';
import { DateTime } from 'luxon';
import { Date2024Oct23 } from '../../fakes/monthDays';

const dateToTest = '2024-02-10';
const mockGenerateCalendar = vi.fn().mockReturnValue(Date2024Oct23);
const currentDate = { value: DateTime.fromISO(dateToTest) };
const monthDays = { value: [] };

vi.mock('@/composables/useRenderCalendar', () => ({
  default: vi.fn(() => ({
    currentDate,
    monthDays,
    generateCalendar: mockGenerateCalendar,
  })),
}));

let wrapper: VueWrapper<InstanceType<typeof ButtonPrevMonth>>

describe('ButtonPrevMonth', () => {

  beforeEach(() => {
    wrapper = mount(ButtonPrevMonth);
  });

  afterEach(() => {
    vi.clearAllMocks()
  })

  it('should render the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('init values for composable useRenderCalendar', () => {
    expect(monthDays.value).toStrictEqual([]);
    expect(currentDate.value).toStrictEqual(DateTime.fromISO(dateToTest));
  })

  it('do click on the button', async () => {
    const expected = '2024-01-10'

    await wrapper.find('button').trigger('click');

    expect(monthDays.value).toStrictEqual(Date2024Oct23);
    expect(currentDate.value).toStrictEqual(DateTime.fromISO(expected));
    expect(mockGenerateCalendar).toHaveBeenCalledWith(expected);
    expect(wrapper.emitted('handle')).toStrictEqual([[expected]]);

    await wrapper.find('button').trigger('click');
    expect(wrapper.emitted('handle')).toStrictEqual([[expected], ['2023-12-10']]);
    expect(currentDate.value).toStrictEqual(DateTime.fromISO('2023-12-10'));
  })
});