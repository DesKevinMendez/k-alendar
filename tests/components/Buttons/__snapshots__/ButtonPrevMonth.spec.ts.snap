// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonPrevMonth > should render the component 1`] = `"<button data-v-11092de0="" type="button" class="k-alendar-navegation-prev">←</button>"`;
